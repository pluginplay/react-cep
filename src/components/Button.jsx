import styled from 'styled-components'

export default styled.button`
  padding: 4px 8px;
  margin: 10px auto;
  min-width: 100px;
  font-family: "${props => props.theme.ae.baseFontFamily}", "Segoe UI", "San Francisco", sans-serif;
  color: white;
  background: ${props => props.theme.background.dark};
  border: solid 1px ${props => props.theme.background.light};
  font-size: ${props => props.theme.ae.baseFontSize}pt;
  outline: none;
  box-sizing: border-box;
  transition: background 0.1s ease-in-out, border-color 0.1s ease-in-out, transform 0.1s ease-in-out, 
    opacity 0.1s ease-in-out;
  opacity: 1;
  transform: scale(1);
  cursor: pointer;
  display: inline-block;
  &:not(:disabled):hover, &:not(:disabled):focus, &:not(:disabled):active {
    border-color: ${props => props.theme.ae.systemHighlightColor};
    background: ${props => props.theme.background.darker};
  }
  &:not(:disabled):active { 
    transform: scale(0.95);
    background: ${props => props.theme.ae.panelBackgroundColor};
  }
  &:disabled {
    cursor: default;
    opacity: 0.5;
  }
  @media (max-width: 320px) {
    padding: 3px 4px;
    margin: 4px auto;
    font-size: 8pt;
  }
`
