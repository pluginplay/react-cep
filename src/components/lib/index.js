import SystemState from '../../models/SystemState'
import PersistedState from '../../models/PersistedState'

export { SystemState, PersistedState }
