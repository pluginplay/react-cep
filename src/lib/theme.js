import * as csInterface from 'cep-lib/csinterface'
import { lighten, darken } from 'polished'

export const getDefaultTheme = () => {
  const ae = csInterface.getThemeInformation()
  ae.systemHighlightColor = '#0078d7' // We have to hard-code this because Windows gets the wrong color.
  return {
    ae,
    background: {
      lighter: lighten(0.2, ae.panelBackgroundColor),
      light: lighten(0.1, ae.panelBackgroundColor),
      dark: darken(0.05, ae.panelBackgroundColor),
      darker: darken(0.1, ae.panelBackgroundColor)
    },
    colors: {
      green: '#23EB87',
      red: '#eb093f',
      blue: ae.systemHighlightColor
    }
  }
}
