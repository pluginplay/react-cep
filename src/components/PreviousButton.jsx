import React from 'react'
import Button from './Button'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import Flex from 'styled-flex-component'
import arrowLeft from '../assets/icons/arrow-left.svg'

class PreviousButton extends React.Component {
  render () {
    return (
      <Button onClick={this.props.onClick} className={this.props.className}>
        <Flex full alignCenter>
          <img src={arrowLeft} alt={'Back'} />
          <span>Back</span>
        </Flex>
      </Button>
    )
  }
}

PreviousButton.propTypes = {
  onClick: PropTypes.func.isRequired
}

export default styled(PreviousButton)`
  background: none;
  border: none;
  min-width: 0;
  img {
    height: 12px;
    margin-right: 5px;
  }
  &:active, &:focus {
    background: none;    
  }
`
