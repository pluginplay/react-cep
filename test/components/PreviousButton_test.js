import { expect, withTheme } from '../setup'
import React from 'react'
import PreviousButton from '../../src/components/PreviousButton'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { getDefaultTheme } from '../../src/lib/theme'

import Button from '../../src/components/Button'

let onClick = null
beforeEach(() => {
  onClick = jest.fn()
})

it('renders correctly', () => {
  const tree = renderer
    .create(withTheme(<PreviousButton onClick={onClick} />))
    .toJSON()
  jestExpect(tree).toMatchSnapshot()
})

it('forwards props', () => {
  const wrapper = shallow(<PreviousButton theme={getDefaultTheme()} onClick={onClick} />).dive()
  expect(wrapper.find(Button).first()).to.have.prop('onClick').equal(onClick)
  expect(wrapper.find(Button).first()).to.have.prop('className')
  expect(wrapper.find('span').first()).to.have.text('Back')
})
