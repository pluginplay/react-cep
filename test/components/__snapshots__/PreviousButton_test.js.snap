// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 {
  padding: 4px 8px;
  margin: 10px auto;
  min-width: 100px;
  font-family: "Adobe Clean","Segoe UI","San Francisco",sans-serif;
  color: white;
  background: #191919;
  border: solid 1px #404040;
  font-size: 10pt;
  outline: none;
  box-sizing: border-box;
  -webkit-transition: background 0.1s ease-in-out,border-color 0.1s ease-in-out,-webkit-transform 0.1s ease-in-out, opacity 0.1s ease-in-out;
  -webkit-transition: background 0.1s ease-in-out,border-color 0.1s ease-in-out,transform 0.1s ease-in-out, opacity 0.1s ease-in-out;
  transition: background 0.1s ease-in-out,border-color 0.1s ease-in-out,transform 0.1s ease-in-out, opacity 0.1s ease-in-out;
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  cursor: pointer;
  display: inline-block;
}

.c1:not(:disabled):hover,
.c1:not(:disabled):focus,
.c1:not(:disabled):active {
  border-color: #0078d7;
  background: #0c0c0c;
}

.c1:not(:disabled):active {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
  background: #262626;
}

.c1:disabled {
  cursor: default;
  opacity: 0.5;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
}

.c0 {
  background: none;
  border: none;
  min-width: 0;
}

.c0 img {
  height: 12px;
  margin-right: 5px;
}

.c0:active,
.c0:focus {
  background: none;
}

<button
  className="c0 c1"
  onClick={[Function]}
>
  <div
    className="c2"
  >
    <img
      alt="Back"
      src="test-file-stub"
    />
    <span>
      Back
    </span>
  </div>
</button>
`;
